embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/SessionApiTest.kt

```diff
@@ -50,20 +50,27 @@ internal class SessionApiTest {
 
             // validate snapshots separately, as the JSON diff is tricky to debug
             val snapshots = checkNotNull(message.data.spanSnapshots)
-            assertEquals(2, snapshots.size)
+            assertEquals(1, snapshots.size)
 
             // validate network status span
             val networkStatusSpan = snapshots.single { it.name == "emb-network-status" }
             assertEquals(startTime, networkStatusSpan.startTimeNanos?.nanosToMillis())
             assertEquals("sys.network_status", networkStatusSpan.attributes?.findAttributeValue("emb.type"))
 
             // validate session span
-            val sessionSpan = snapshots.single { it.name == "emb-session" }
+            val spans = checkNotNull(message.data.spans)
+            val sessionSpan = spans.single { it.name == "emb-session" }
             assertEquals(startTime, sessionSpan.startTimeNanos?.nanosToMillis())
             assertNotNull(sessionSpan.attributes?.findAttributeValue(SessionIncubatingAttributes.SESSION_ID.key))
-            val attrs = checkNotNull(sessionSpan.attributes?.filterNot {
+            val attrs = checkNotNull(sessionSpan.attributes)
```

Comment:
A lot more attributes are added to a session span that actually ends. Fully validate the ones were the value are deterministic, and check that the expected keys exist with values that either aren't constant or is immaterial to verify